src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,142 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private T[] array;
+    private int arraySize = 0;
```

Comment:
The name 'arraySize' could be more informative. Consider renaming it to 'size' or 'currentSize' to better reflect its purpose.